warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PullupUnexecutableNodesVisitor.java

```diff
@@ -242,9 +242,20 @@ public Object visit(ASTNotNode node, Object data) {
     public Object visit(ASTReference node, Object data) {
         // if a delayed predicate, then change it to a regular reference
         if (ASTDelayedPredicate.instanceOf(node)) {
-            JexlNode source = ASTDelayedPredicate.getDelayedPredicateSource(node);
-            JexlNodes.swap(node.jjtGetParent(), node, source);
-            return source;
+            
+            // continue to pull up delayed markers in the odd case that a double delay occurred
+            JexlNode marker = node;
+            while (ASTDelayedPredicate.instanceOf(marker)) {
+                if (node.jjtGetNumChildren() > 1) {
+                    log.warn("Tried to pull up a delayed marker that had more than one child.");
+                }
+                
+                // This swap should be safe, the reference node should have a single child that is a reference expression
```

Comment:
You check, but you have not guaranteed that the reference node has a single child.  Consider throwing an exception if you encounter a reference node with multiple children, or just remove the check & logging since this should never happen.